sdk/search/search/review/search.api.md

```diff
@@ -131,13 +130,13 @@ export class SearchIndexClient<T> {
     readonly apiVersion: string;
     autocomplete<Fields extends keyof T>(options: AutocompleteOptions<Fields>): Promise<AutocompleteResult>;
     count(options?: CountOptions): Promise<number>;
-    deleteDocuments(keyName: string, keyValues: string[], options?: DeleteDocumentsOptions): Promise<IndexDocumentsResult>;
+    deleteDocuments(keyName: keyof T, keyValues: string[], options?: DeleteDocumentsOptions): Promise<IndexDocumentsResult>;
     readonly endpoint: string;
     getDocument<Fields extends keyof T>(key: string, options?: GetDocumentOptions<Fields>): Promise<T>;
     readonly indexName: string;
     listSearchResults<Fields extends keyof T>(options?: SearchOptions<Fields>): SearchIterator<Pick<T, Fields>>;
-    modifyIndex(batch: IndexAction[], options?: ModifyIndexOptions): Promise<IndexDocumentsResult>;
-    suggest<Fields extends keyof T>(options: SuggestOptions<Fields>): Promise<SuggestDocumentsResult<Pick<T, Fields>>>;
+    modifyIndex(batch: IndexAction<T>[], options?: ModifyIndexOptions): Promise<IndexDocumentsResult>;
+    suggest<Fields extends keyof T = never>(options: SuggestOptions<Fields>): Promise<SuggestDocumentsResult<Pick<T, Fields>>>;
     updateDocuments(documents: T[], options?: UpdateDocumentsOptions): Promise<IndexDocumentsResult>;
```

Comment:
`uploadDocuments` and `updateDocuments` are extremely visually similar. We ought to use the same terminology as the REST API, because the semantics do not fit the typical "create, insert, update, upsert" model. "upload" can be used for create or update, but whole-document update rather than patching. "merge" is just patching. "mergeOrUpload" is like "upload if not exists, merge otherwise". "update" doesn't communicate these differences, and it looks like we're missing a case.

Reply:
Renamed `updateDocuments` to `modifyDocuments` in an attempt to help. Does this sound better to communicate the nuance of merge?

Reply:
`mergeDocuments` as a name is also possible, but to me that seems to connote merging documents on the server rather than merging updates to documents

Reply:
I'm not sure I understand the distinction between those last two. `modify` is ok, but I'd still prefer to keep the terminology we've already established.

Reply:
okay, I'll rename this to `merge`, better not to invent new verbs